tests/portstat/test_portstat.py

```diff
@@ -0,0 +1,197 @@
+
+    logger.info('Verify that the file names are not in the /tmp directory')
+    for stats_file in stats_files:
+        pytest_assert(not duthost.stat(path='/tmp/portstat-{uid}/{uid}-{filename}'\
+                      .format(uid=uid, filename=stats_file))['stat']['exists'])
+
+
+@pytest.mark.parametrize('command',
+                         ['portstat -d -t', 'portstat -d --tag', 'portstat --delete -t', 'portstat --delete --tag'])
+def test_portstat_delete_tag(duthost, command):
+
+    stats_files = ('test_1', 'test_2', 'test_delete_me')
+    file_to_delete = stats_files[2]
+    files_not_deleted = stats_files[:2]
+
+    logger.info('Create several test stats files')
+    for stats_file in stats_files:
+        duthost.command('portstat -c -t {}'.format(stats_file))
+
+    logger.info('Verify that the file names are in the /tmp directory')
+    uid = duthost.command('id -u')['stdout'].strip()
+    for stats_file in stats_files:
+        pytest_assert(duthost.stat(path='/tmp/portstat-{uid}/{uid}-{filename}'\
+                      .format(uid=uid, filename=stats_file))['stat']['exists'])
+
+    full_delete_command = command + ' ' + file_to_delete
+    logger.info('Run the command to be tested "{}"'.format(full_delete_command))
+    duthost.command(full_delete_command)
+
+    logger.info('Verify that the deleted file name is not in the directory')
+    pytest_assert(not duthost.stat(path='/tmp/portstat-{uid}/{uid}-{filename}'\
+                  .format(uid=uid, filename=file_to_delete))['stat']['exists'])
+
+    logger.info('Verify that the remaining file names are in the directory')
+    for stats_file in files_not_deleted:
+        pytest_assert(duthost.stat(path='/tmp/portstat-{uid}/{uid}-{filename}'\
+                      .format(uid=uid, filename=stats_file))['stat']['exists'])
+
+
+@pytest.mark.parametrize('command', ['portstat -a', 'portstat --all'])
+def test_portstat_display_all(duthost, command):
+
+    base_portstat = parse_portstat(duthost.command('portstat')['stdout_lines'])
+    all_portstats = parse_portstat(duthost.command(command)['stdout_lines'])
+    pytest_assert(base_portstat and all_portstats, 'No parsed command output')
+
+    logger.info('Verify the all number of columns is greater than the base number of columns')
+    for intf in all_portstats.keys():
+        pytest_assert(len(all_portstats[intf].keys()) > len(base_portstat[intf].keys()))
+
+
+@pytest.mark.parametrize('command', ['portstat -p 1', 'portstat --period 1'])
+def test_portstat_period(duthost, command):
+
+    output = duthost.command(command)
+    pytest_assert('The rates are calculated within 1 seconds period' in output['stdout_lines'][0])
+
+
+@pytest.mark.parametrize('command', ['portstat -h', 'portstat --help', 'portstat', 'portstat -v',
+                                     'portstat --version', 'portstat -j', 'portstat --json',
+                                     'portstat -r', 'portstat --raw'])
+def test_portstat_other(duthost, command):
```

Comment:
Do you think something like `test_portstat_no_exceptions` be a clearer name for this?